lib/tcp/tcp_packet.ml

```diff
@@ -73,7 +73,6 @@ module Marshal = struct
   let unsafe_fill ~pseudoheader ~payload t buf options_len =
     let data_off = (sizeof_tcp / 4) + (options_len / 4) in
     let buf = Cstruct.sub buf 0 (data_off * 4) in
-    Cstruct.memset buf 0; (* autopad *)
```

Comment:
Don't see how zeroing policy change could affect it since as long as it was doing a memset it would always blow away the options that had been set before this call.  Of course if it was not doing a memset then it would explain the change.  And also confirm that this line is not required.  I don't have a test, if you can think of a test for this please do add.  Thx.